bin/oneoff/unsent_virtual_pd_workshop_emails.rb

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env ruby
+
+require_relative '../../dashboard/config/environment'
+
+workshops = Pd::Workshop.future.
+  where("pd_workshops.properties LIKE '%\"virtual\":true%'").
```

Comment:
I'm sure this works, but something like this might be easier in the future (not trying to be pedantic, but I think Ruby is new to you yeah?)

`Pd::Workshop.future.filter {|workshop| workshop.virtual && ['CS Discoveries', 'CS Principles'].include? workshop.course}`

Reply:
This is probably worse, never mind :).

Reply:
ahhh, i was looking for something like that, thank you!

Reply:
i was trying to do some fancy stuff with joins in active record, but i think there probably aren't enough workshops for it to matter too much trying to make it optimized for speed

Reply:
Ahh yeah good call, probably the raw SQL version is way faster. Also like you said probably doesn't matter for this case. Another good thing to know for when needing to iterate through a lot of records is `find_each`:

https://api.rubyonrails.org/v6.1.3.1/classes/ActiveRecord/Batches.html#method-i-find_each